src/libs/ajax/AzureStorage.ts

```diff
@@ -111,6 +111,40 @@ export const AzureStorage = (signal?: AbortSignal) => ({
     }), notebooks)
   },
 
+  blobMetadata: (azureStorageUrl: string) => {
+    const getObjectMetadata = async () => {
+      // assumption is made that container name guid in uri always matches the workspace Id guid it is present in
+      const workspaceId = azureStorageUrl.split('/')[3].replace('sc-', '')
+      const fileName = _.last(azureStorageUrl.split('/'))?.split('.').join('.')
+
+      try {
+        const { sas: { token } } = await AzureStorage(signal).details(workspaceId)
+
+        // instead of taking the url returned by azure storage, take it from the incoming url since there may be a folder path
+        const urlwithFolder = new URL(azureStorageUrl)
+        const azureSasStorageUrl = `https://${urlwithFolder.hostname}${urlwithFolder.pathname}?&${token}`
+
+        const res = await fetchOk(azureSasStorageUrl)
```

Comment:
If I understand correctly - and from what I see in dev tools - this line downloads the entire file in question, then looks at the headers returned by that response. This feels very wasteful and will cause latency problems and potential browser-memory issues when users are working with large, multi-GB files.

Is there a HEAD request we could make instead of a full GET request that gives the same information? If not a HEAD request, can we do a GET but specify a very small [range](https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests)? Or some other Azure Storage API we can call? All we need is the `last-modified` and `content-length` values.

Reply:
aha! I swear I tried the HEAD operation but couldn't get it to work....not sure what was wrong but now it does! 

we technically also need the Created datetime, which should be in the headers but is not, spent a good 30 mins of it with no progress, will table for later.